redis/pool_test.go

```diff
@@ -48,21 +48,46 @@ func (c *poolTestConn) Close() error {
 func (c *poolTestConn) Err() error { return c.err }
 
 func (c *poolTestConn) Do(commandName string, args ...interface{}) (interface{}, error) {
+	return c.do(c.Conn.Do, commandName, args...)
+}
+
+func (c *poolTestConn) DoContext(ctx context.Context, commandName string, args ...interface{}) (interface{}, error) {
+	cwt, ok := c.Conn.(redis.ConnWithContext)
```

Comment:
I know this is what the rest of the code base has but it's actually a cut ant past error from ConnWithTimeout, can we change this to use `cwc` so it correctly represents the actual type.
```suggestion
	cwc, ok := c.Conn.(redis.ConnWithContext)
```